apps/segment/src/modules/webhooks/webhook-activity/webhook-activity-service.test.ts

```diff
@@ -0,0 +1,92 @@
+    const mockedClient: IWebhooksActivityClient = {
+      enableSingleWebhook: vi.fn(() => Promise.resolve(ok(undefined))),
+      fetchAppWebhooksInformation: vi.fn(() =>
+        Promise.resolve(
+          ok([
+            { id: "webhook-id-1", isActive: true },
+            { id: "webhook-id-2", isActive: true },
+          ]),
+        ),
+      ),
+    };
+
+    const service = new WebhookActivityService("app-id", mockedClient);
+
+    await service.enableAppWebhooks();
+
+    expect(mockedClient.fetchAppWebhooksInformation).toHaveBeenCalled();
+    expect(mockedClient.enableSingleWebhook).toHaveBeenCalledTimes(2);
+    expect(mockedClient.enableSingleWebhook).toHaveBeenNthCalledWith(1, "webhook-id-1");
+    expect(mockedClient.enableSingleWebhook).toHaveBeenNthCalledWith(2, "webhook-id-2");
+  });
+
+  it("should get information if webhooks are active", async () => {
+    const mockedClient: IWebhooksActivityClient = {
+      enableSingleWebhook: vi.fn(() => Promise.resolve(ok(undefined))),
+      fetchAppWebhooksInformation: vi.fn(() =>
+        Promise.resolve(
+          ok([
+            { id: "webhook-id-1", isActive: true },
+            { id: "webhook-id-2", isActive: false },
+          ]),
+        ),
+      ),
+    };
+
+    const service = new WebhookActivityService("app-id", mockedClient);
+
+    const response = await service.getWebhooksIsActive();
+
+    expect(response._unsafeUnwrap()).toStrictEqual([true, false]);
+  });
+
+  it("should throw error when fetching webhooks information fails", async () => {
+    const mockedClient: IWebhooksActivityClient = {
+      enableSingleWebhook: vi.fn(),
+      fetchAppWebhooksInformation: vi.fn(() =>
+        Promise.resolve(err("Error during fetching webhooks information")),
+      ),
+    };
+
+    const service = new WebhookActivityService("app-id", mockedClient);
+
+    // TODO: I used try-catch because I couldn't make the expect statement work
+    try {
+      await service.enableAppWebhooks();
+    } catch (e) {
+      expect(e).toBeInstanceOf(WebhookActivityService.WebhookActivityServiceWebhooksError);
+    }
+
+    expect(mockedClient.fetchAppWebhooksInformation).toHaveBeenCalled();
+    expect(mockedClient.enableSingleWebhook).not.toHaveBeenCalled();
+  });
+
+  it("should throw error when enabling app webhooks fails", async () => {
+    const mockedClient: IWebhooksActivityClient = {
+      enableSingleWebhook: vi.fn(() => Promise.resolve(err("Error during enabling webhooks"))),
+      fetchAppWebhooksInformation: vi.fn(() =>
+        Promise.resolve(ok([{ id: "webhook-id", isActive: false }])),
+      ),
+    };
+
+    const service = new WebhookActivityService("app-id", mockedClient);
+
+    // TODO: I used try-catch because I couldn't make the expect statement work
+    try {
+      await service.enableAppWebhooks();
+    } catch (e) {
+      expect(e).toBeInstanceOf(WebhookActivityService.WebhookActivityServiceWebhooksError);
+    }
```

Comment:
question: didn't this work?

```ts
await expect(service.enableAppWebhooks).throws.toBeInstanceOf(...);
```

If for some reason it doesn't and we need try...catch then maybe let's also add expect in the try {} block so that we know if test failed (otherwise it will silently pass)

```ts
try {
  await service.enableAppWebhooks();
  expect(false); // this shouldn't be called since exception will be thrown
} catch (e)
```

Reply:
This is leftover from previous implementation - now `enableAppWebhooks` will return `err | ok` from neverthrow - I will update the tests